### ISSUE TITLE
Optional chaining for `Array.push` throws error after upgrade to 1.4.3

### ISSUE THREAD

**Issue by Sebaestschjin** [TypeScriptToLua/TypeScriptToLua#1268]
After upgrading to version 1.4.3, the following example code throws an error:

```ts
interface Y {
  elements?: number[];
}

const x: Y = {};
x.elements?.push(1);
```

This did work in 1.4.2. 

I only encountered this using the `push` function of `Array`. Chaining other functions from `Array` worked fine as well as using other types for chaining, like `string`.

The error message is:
	
```
TypeError: Cannot read properties of undefined (reading 'kind')
    at Object.isOuterExpression (<node-modules-dir>/typescript@4.6.2\node_modules\typescript\lib\typescript.js:29305:22)
    at findFirstNonOuterParent (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\utils\typescript\index.js:43:15)
    at transformSingleElementArrayPush (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\builtins\array.js:36:97)
    at transformArrayPrototypeCall (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\builtins\array.js:65:28)
    at transformBuiltinCallExpression (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\builtins\index.js:122:56)
    at Object.transformCallExpression [as transform] (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\visitors\call.js:162:90)
    at TransformationContext.transformNode (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\context\context.js:48:67)    
    at TransformationContext.transformExpression (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\context\context.js:60:31)
    at <node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\visitors\optional-chaining.js:105:30
    at transformInPrecedingStatementScope (<node-modules-dir>/typescript-to-lua@1.4.3_typescript@4.6.2\node_modules\typescript-to-lua\dist\transformation\utils\preceding-statements.js:6:35)
```

**Comment by Perryvw** [comment]
This is already fixed but not yet released, I will do a patch release later today to get this fix out. Duplicate of #1256

**Comment by Perryvw** [comment]
Should be fixed with 1.4.4
